lib/iris/experimental/ugrid.py

```diff
@@ -1680,102 +1679,316 @@ def remove(
+        for parameter, location_name in (
+            (node, "node"),
+            (edge, "edge"),
+            (face, "face"),
+        ):
+            members = location_filter(members, parameter, location_name)
+
+        # No need to actually modify filtering behaviour - already won't return
+        # any face cf-roles if none are present.
+        if face and self.NDIM < 2:
+            message = (
+                "Ignoring request to filter for non-existent 'face' cf-roles."
+            )
+            logger.debug(message, extra=dict(cls=self.__class__.__name__))
+
+        result = filter_cf(
+            members,
+            item=item,
+            standard_name=standard_name,
+            long_name=long_name,
+            var_name=var_name,
+            attributes=attributes,
+        )
+
+        # Use the results to filter the _members dict for returning.
+        result_ids = [id(r) for r in result]
+        result_dict = {
+            k: v for k, v in self._members.items() if id(v) in result_ids
+        }
+        return result_dict
+
+    def remove(
+        self,
+        item=None,
+        standard_name=None,
+        long_name=None,
+        var_name=None,
+        attributes=None,
+        cf_role=None,
+        node=None,
+        edge=None,
+        face=None,
+    ):
+        removal_dict = self.filters(
+            item=item,
+            standard_name=standard_name,
+            long_name=long_name,
+            var_name=var_name,
+            attributes=attributes,
+            cf_role=cf_role,
+            node=node,
+            edge=edge,
+            face=face,
+        )
+        for cf_role in self.REQUIRED:
+            not_removed = removal_dict.pop(cf_role, None)
+            if not_removed:
+                message = (
+                    f"Ignoring request to remove required connectivity "
+                    f"({cf_role}: {not_removed!r})"
+                )
+                logger.debug(message, extra=dict(cls=self.__class__.__name__))
+
+        for cf_role in removal_dict.keys():
+            self._members[cf_role] = None
+
+        return removal_dict
+
+
+class _Mesh1DConnectivityManager(_MeshConnectivityManagerMixin):
+    REQUIRED = ("edge_node_connectivity",)
+    OPTIONAL = ()
+    NDIM = 1
+
+    def __init__(self, *connectivities):
+        super().__init__(*connectivities)
+
+    @property
+    def all_members(self):
+        return Mesh1DConnectivities(edge_node=self.edge_node)
+
+    @property
+    def edge_node(self):
+        return self._members["edge_node_connectivity"]
+
+
+class _Mesh2DConnectivityManager(_MeshConnectivityManagerMixin):
+    REQUIRED = ("face_node_connectivity",)
+    OPTIONAL = (
+        "edge_node_connectivity",
+        "face_edge_connectivity",
+        "face_face_connectivity",
+        "edge_face_connectivity",
+        "boundary_node_connectivity",
+    )
+    NDIM = 2
+
+    def __init__(self, *connectivities):
+        super().__init__(*connectivities)
```

Comment:
@trexfeathers Again, you can remove this `__init__` :+1:

Reply:
Nice catch! A classic product of many local refactors. It's obvious now you've said it.

a8897abb48759742a48b84f8f82986c78bdb42b6